core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -745,4 +774,19 @@ private void ensureExists(AlluxioURI path) throws IOException {
       throw new IOException(e);
     }
   }
+
+  /**
+   * Returns original path if shim mode is enabled.
+   * Returns URI path component for when shim mode is not enabled.
+   *
+   * @param path the input path
+   * @return the Alluxio path
+   */
+  private String getAlluxioPath(Path path) {
```

Comment:
not sure I understand why both AbstractFileSystem and RetryHandlingFileSystemMasterClient need to convert paths to string representations. does the first strip out the scheme for an alluxio path only? whereas the second conversion is more of a performance optimization?

Reply:
`AbstractFileSystem` will either pass the `Path` as full path or it'll strip scheme&authority from it. This is something that `AbstractFileSystem` knows based on whether it's a shim or not. 

`RetryHandlingFileSystemMasterClient` converts it to because paths are simple strings on transport layer. 

Also, `AbstractFileSystem` converts from `Path`, while `RetryHandlingFileSystemMasterClient` converts from `AlluxioURI`.